package/ports_daemon/ports_daemon/src/ports.c

```diff
@@ -364,6 +398,33 @@ static u8 protocol_index_to_mode(int protocol_index)
   return protocol_index + 1;
 }
 
+static int blacklisted_mode_lookup(u8 mode, const port_config_t *port_config)
+{
+  assert(mode <= MAX_NUM_PROTOCOLS);
+  int mode_index = mode_to_protocol_index(mode);
+  assert(port_config->blacklisted_indices[mode_index] != 1);
+
+  int blacklisted_counter = 0;
+  for (int i = 0; i < mode_index; i++) {
+    if (port_config->blacklisted_indices[i] == 1) {
+      blacklisted_counter++;
+    }
+  }
+
+  return protocol_index_to_mode(mode_index - blacklisted_counter);
+}
+
+static int blacklisted_index_lookup(int index, const port_config_t *port_config)
+{
```

Comment:
I have no idea what this is doing.  Is this trying to fix the offset of the index when going from blacklisted index back to a regular index?  Where the offset is dependent on the number blacklisted index that appear before that index?

Reply:
Ben and I did a pair programming session where we went through and refactored the code.
This function was removed - as you noted, it wasn't really doing anything useful. 

The new commit message should (hopefully) explain the new implementation, which is a lot simpler.